linalg/sparsemat.cpp

```diff
@@ -1496,9 +1496,13 @@ void SparseMatrix::Symmetrize()
 
 int SparseMatrix::NumNonZeroElems() const
 {
-   if (A != NULL)  // matrix is finalized
+
+   if ( Finalized() )
    {
-      return I[height];
+      Memory<int> nnz_mem(I, height, 1);
+      const int nnz = *nnz_mem.Read(MemoryClass::HOST, 1);
+      nnz_mem.Delete();
+      return nnz;
```

Comment:
Fixing this will require some nontrivial changes to the memory manager.

I suggest for now doing the full `HostRead` to get the `nnz`, and the more efficient version can be added as a separate PR (unrelated to the `[CUDA]` unit tests).

Reply:
Done in [a621595](https://github.com/mfem/mfem/pull/2892/commits/a621595c8cdea70b82a691a2bf642486a8a35ed9)